node/subsystem-util/src/inclusion_emulator/staging.rs

```diff
@@ -760,6 +765,16 @@ fn validate_against_constraints(
 		))
 	}
 
+	if modifications.dmp_messages_processed == 0 {
+		if constraints
+			.dmp_remaining_messages
+			.get(0)
+			.map_or(false, |&msg_sent_at| msg_sent_at <= relay_parent.number)
+		{
+			return Err(FragmentValidityError::DmpAdvancementRule)
+		}
+	}
+
```

Comment:
This check only makes sense in the context of some candidate since it depends on the relay parent number. Thus, it's a part of Fragment-checks, not constraints